### DISCUSSION TITLE
Convert SpatialExperiment Object to Seurat Object

### DISCUSSION THREAD

**Discussion by cathalgking** [satijalab/seurat#6918] [Q&A]
I have a 10x Visium experiment that was aligned with Space Ranger. There is 4 images/samples and each output contains the standard ''/outs/'' folder with image, matrix, tissue positions etc. I read the data into a SpatialExperiment (spe) with the below code.

1 - I would like to convert my SpatialExperiment object to a Seurat object for some downstream analyses. I know that there is functionality to convert a SingleCellExperiment object to a Seurat object with as.SingleCellExperiment(x) but does something similar exist for a Spatial experiment conversion to a Seurat object? When I try to use that function on my spe I get the Error shown below.
2 - Or, is there a way to read in my multiple image/sample experiment into a Seurat object at once as is done with the read10xVisium() function below for a SpatialExperiment?

```{r}
samples <- c("~/A1count/outs",
             "~/B1count/outs",
             "~/C1count/outs",
             "~/D1count/outs")

sample_ids = c("A1count", "B1count", "C1count", "D1count")

spe <- read10xVisium(
  samples = samples,
  sample_id = sample_ids,
  type = "sparse",
  data = "raw",
  images = "lowres",
  load = TRUE
)
```
Error in UseMethod(generic = "as.SingleCellExperiment", object = x) : no applicable method for 'as.SingleCellExperiment' applied to an object of class "c('SpatialExperiment', 'SingleCellExperiment', 'RangedSummarizedExperiment', 'SummarizedExperiment', 'RectangularData', 'Vector', 'Annotated', 'vector_OR_Vector')"

I posted the same on bioconductor but not sure if the Seurat package was picked up in the tags.
https://support.bioconductor.org/p/9149216/

**Comment by cathalgking** [comment]
I tried using the merge() function to merge 4 images (samples) together by repeating the merge by adding an image each time - from a 10x Visium slide with 4 capture areas on the slide. Calling the seu object says that there are 4 images within and when calling SpatialDimPlot(seu), the correct 4 images are shown. Are there any issues with this approach? 

![Uploading Seu_merge.png…]()

**Reply by BenjaminDEMAILLE** [reply]
https://satijalab.org/seurat/reference/load10x_spatial

**Reply by BenjaminDEMAILLE** [reply]
Just loop over your samples and merge the Seurat object list
